gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsBatch.java

```diff
@@ -0,0 +1,351 @@
+   * be used to restrict the fields returned or provide page size limits in the same way as for
+   * {@link Dns#listZones(Dns.ZoneListOption...)}. The returned {@link DnsBatchResult} will return a
+   * page of zones upon calling {@link DnsBatchResult#get()} on successful completion, or it will
+   * throw a {@link DnsException} if the operation failed.
+   */
+  public DnsBatchResult<Page<Zone>> listZones(Dns.ZoneListOption... options) {
+    DnsBatchResult<Page<Zone>> result = new DnsBatchResult<>();
+    Map<DnsRpc.Option, ?> optionMap = DnsImpl.optionMap(options);
+    RpcBatch.Callback<ManagedZonesListResponse> callback = createListZonesCallback(result,
+        optionMap);
+    batch.addListZones(callback, optionMap);
+    return result;
+  }
+
+  /**
+   * Adds a request representing the "create zone" operation to this batch. The {@code options} can
+   * be used to restrict the fields returned in the same way as for {@link Dns#create(ZoneInfo,
+   * Dns.ZoneOption...)}. The returned {@link DnsBatchResult} will return the created {@link Zone}
+   * upon calling {@link DnsBatchResult#get()} on successful completion, or it will throw a {@link
+   * DnsException} if the operation failed.
+   */
+  public DnsBatchResult<Zone> createZone(ZoneInfo zone, Dns.ZoneOption... options) {
+    DnsBatchResult<Zone> result = new DnsBatchResult<>();
+    RpcBatch.Callback<ManagedZone> callback = createZoneCallback(this.options, result);
+    Map<DnsRpc.Option, ?> optionMap = DnsImpl.optionMap(options);
+    batch.addCreateZone(zone.toPb(), callback, optionMap);
+    return result;
+  }
+
+  /**
+   * Adds a request representing the "delete zone" operation to this batch. The returned {@link
+   * DnsBatchResult} will return {@code true} upon calling {@link DnsBatchResult#get()} on
+   * successful deletion, {@code false} if the zone was not found, or it will throw a {@link
+   * DnsException} if the operation failed.
+   */
+  public DnsBatchResult<Boolean> deleteZone(String zoneName) {
+    DnsBatchResult<Boolean> result = new DnsBatchResult<>();
+    RpcBatch.Callback<Void> callback = createDeleteZoneCallback(result);
+    batch.addDeleteZone(zoneName, callback);
+    return result;
+  }
+
+  /**
+   * Adds a request representing the "get zone" operation to this batch. The {@code options} can be
+   * used to restrict the fields returned in the same way as for {@link Dns#getZone(String,
+   * Dns.ZoneOption...)}. The returned {@link DnsBatchResult} will return the requested {@link Zone}
+   * upon calling {@link DnsBatchResult#get()} on successful completion, {@code null} if no such
+   * zone exists, or it will throw a {@link DnsException} if the operation failed.
+   */
+  public DnsBatchResult<Zone> getZone(String zoneName, Dns.ZoneOption... options) {
+    DnsBatchResult<Zone> result = new DnsBatchResult<>();
+    RpcBatch.Callback<ManagedZone> callback = createZoneCallback(this.options, result);
```

Comment:
No, this is a joint prep callback method for `getZone` and `createZone`, so we have generic `createZoneCallback`.